Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/UpdateMetadataImportDialogSequence.java

```diff
@@ -0,0 +1,216 @@
+ * Finally, the metadata comparison dialog allows selecting the specific metadata information
+ * imported for a logical structure element.</p>
+ */
+@Named("UpdateMetadataImportDialogSequence")
+@ViewScoped
+public class UpdateMetadataImportDialogSequence implements Serializable {
+
+    private static final Logger logger = LogManager.getLogger(UpdateMetadataImportDialogSequence.class);
+
+    @Inject
+    private CreateProcessForm createProcessForm;
+
+    @Inject
+    private DataEditorForm dataEditorForm;
+
+    @Inject
+    private UpdateMetadataDialog updateMetadataDialog;
+
+    /** 
+     * Is called when a user clicks on the import metadata button of the logical metadata panel.
+     */
+    public void onImportMetadataClick() {
+        if (canImportMetadata()) {
+            importMetadata();
+        }
+    }
+
+    /**
+     * Return true if any logical division is selected such that catalog metadata can be imported.
+     * 
+     * @return true if a logical division is selected
+     */
+    public boolean canImportMetadata() {
+        TreeNode selectedNode = dataEditorForm.getStructurePanel().getSelectedLogicalNodeIfSingle();
+        return Objects.nonNull(selectedNode)
+                && selectedNode.getData() instanceof StructureTreeNode
+                && ((StructureTreeNode) selectedNode.getData()).getDataObject() instanceof LogicalDivision;
+    }
+
+    /**
+     * Open catalog search dialog when the user requests to import metadata from a catalog record.
+     */
+    public void importMetadata() {
+        int templateId = dataEditorForm.getProcess().getTemplate().getId();
+        int projectId = dataEditorForm.getProcess().getProject().getId();
+
+        String previouslySelectedField = createProcessForm.getCatalogImportDialog().getSelectedField();
+        ImportConfiguration previouslySelectedImportConfiguration = createProcessForm.getCurrentImportConfiguration();
+
+        createProcessForm.prepareProcess(templateId, projectId, null, null, false);
+
+        // recover previously selected import configuration and search field
+        if (Objects.nonNull(previouslySelectedImportConfiguration)) {
+            createProcessForm.setCurrentImportConfiguration(previouslySelectedImportConfiguration);
+        }
+        if (Objects.nonNull(previouslySelectedField) && !previouslySelectedField.isEmpty()) {
+            createProcessForm.getCatalogImportDialog().setSelectedField(previouslySelectedField);
+        }
+
+        // always reset search term to empty value
+        createProcessForm.getCatalogImportDialog().setSearchTerm("");
+
+        PrimeFaces.current().executeScript("PF('catalogSearchDialog').show();");
+    }
+
+    /**
+     * Perform a catalog search and open either the histlist dialog (if there are multiple search results) 
+     * or the metadata comparison dialog, if there is only a single search result.
+     */
+    public void catalogSearch() {
```

Comment:
My IDE falsely marks this method and `selectRecord` as unused. Perhaps it would make things clearer for developers in the future if we add a comment that this method and `selectRecord` are indeed not unused, but that their method names are passed to in the `metadataEditor` and `processFromTemplate` templates?